internal/driver/webhtml.go

```diff
@@ -0,0 +1,555 @@
+  var touchGap = function(t1, t2) {
+    var dx = t1.clientX - t2.clientX
+    var dy = t1.clientY - t2.clientY
+    return Math.sqrt(Math.pow(dx, 2) + Math.pow(dy, 2))
+  }
+
+  var handleTouchStart = function(e) {
+    if (mode == IDLE && e.changedTouches.length == 1) {
+      // Start touch based panning
+      var t = e.changedTouches[0]
+      setMode(TOUCHPAN)
+      touchid = t.identifier
+      panStart(t.clientX, t.clientY)
+      e.preventDefault()
+    } else if (mode == TOUCHPAN && e.touches.length == 2) {
+      // Start pinch zooming
+      setMode(TOUCHZOOM)
+      var t1 = e.touches[0]
+      var t2 = e.touches[1]
+      touchid = t1.identifier
+      touchid2 = t2.identifier
+      initScale = currentScale
+      initGap = touchGap(t1, t2)
+      centerPoint = toSvg((t1.clientX + t2.clientX) / 2,
+                          (t1.clientY + t2.clientY) / 2)
+      e.preventDefault()
+    }
+  }
+
+  var handleTouchMove = function(e) {
+    if (mode == TOUCHPAN) {
+      var t = findTouch(e.changedTouches, touchid)
+      if (t == null) return
+      if (e.touches.length != 1) {
+        setMode(IDLE)
+        return
+      }
+      panMove(t.clientX, t.clientY)
+      e.preventDefault()
+    } else if (mode == TOUCHZOOM) {
+      // Get two touches; new gap; rescale to ratio.
+      var t1 = findTouch(e.touches, touchid)
+      var t2 = findTouch(e.touches, touchid2)
+      if (t1 == null || t2 == null) return
+      var gap = touchGap(t1, t2)
+      rescale(initScale * gap / initGap, centerPoint)
+      e.preventDefault()
+    }
+  }
+
+  var handleTouchEnd = function(e) {
+    if (mode == TOUCHPAN) {
+      var t = findTouch(e.changedTouches, touchid)
+      if (t == null) return
+      panMove(t.clientX, t.clientY)
+      setMode(IDLE)
+      e.preventDefault()
+      if (!moved) clickHandler(t.target)
+    } else if (mode == TOUCHZOOM) {
+      setMode(IDLE)
+      e.preventDefault()
+    }
+  }
+
+  svg.addEventListener("mousedown", handleScanStart)
+  svg.addEventListener("mouseup", handleScanEnd)
+  svg.addEventListener("touchstart", handleTouchStart)
+  svg.addEventListener("touchmove", handleTouchMove)
+  svg.addEventListener("touchend", handleTouchEnd)
+  svg.addEventListener("wheel", handleWheel, true)
+}
+
+function dotviewer(nodes) {
+  // Elements
+  var detailsButton = document.getElementById("details")
+  var detailsText = document.getElementById("detailtext")
+  var listButton = document.getElementById("list")
+  var disasmButton = document.getElementById("disasm")
+  var resetButton = document.getElementById("reset")
+  var focusButton = document.getElementById("focus")
+  var showButton = document.getElementById("show")
+  var ignoreButton = document.getElementById("ignore")
+  var hideButton = document.getElementById("hide")
+  var search = document.getElementById("searchbox")
+  var graph0 = document.getElementById("graph0")
+  var svg = graph0.parentElement
+
+  var currentRe = ""
```

Comment:
Switched to const where possible, let elsewhere.
Switched to "for..of" where possible.
I didn't spot any place where lambdas or destructuring assingment would make things easier to read.